### ISSUE TITLE
Signing does not work on a ledger wallet

### ISSUE THREAD

**Issue by Inphi** [ethereum-optimism/superchain-ops#392]
**Bug Description**

I'm on unable to sign a nested safe transaction for task execution(e.g. [this](https://github.com/ethereum-optimism/superchain-ops/pull/390)).

```
❯ just --dotenv-path $(pwd)/.env  --justfile ../../../nested.just sign council

Running script with assertions
Using script /Users/inphi/optimism/superchain-ops/tasks/sep/021-stage-1-4/NestedSignFromJson.s.sol
Using council safe at 0xf64bc17485f0B4Ea5F06A96514182FC4cB561977
getting signer address...
Error:
Could not connect to Ledger device.
Make sure it's connected and unlocked, with no other desktop wallet apps open.

Context:
- received an unexpected empty response
Signing with:

[⠊] Compiling...
No files changed, compilation skipped
2024/12/05 19:10:27 Warning: signer creation failed: error deriving ledger account (have you unlocked?): reply lacks public key entry
Running 'forge script /Users/inphi/optimism/superchain-ops/tasks/sep/021-stage-1-4/NestedSignFromJson.s.sol --rpc-url https://ethereum-sepolia.publicnode.com --sig signJson(string,address) /Users/inphi/optimism/superchain-ops/tasks/sep/021-stage-1-4/input.json 0xf64bc17485f0B4Ea5F06A96514182FC4cB561977
No files changed, compilation skipped
```

This seems to be a more general bug in both `eip712sign` and `cast` on macos:
```
❯ eip712sign --address --hd-paths "m/44'/60'/0'/0/0" --ledger            12s 09:33:31
2024/12/06 09:33:32 Warning: signer creation failed: error deriving ledger account (have you unlocked?): reply lacks public key entry

❯ cast wallet address --ledger --mnemonic-derivation-path "m/44'/60'/0'/0/0"
Error:
Could not connect to Ledger device.
Make sure it's connected and unlocked, with no other desktop wallet apps open.

Context:
- received an unexpected empty response
```

**Debugging**

A recent change in behavior that I've noticed is that I'm required to launch the "Ethereum" app on my ledger wallet in order to interact with the ethereum account it using various CLI tools. This is a requirement to use the `ledger-live` CLI and also `cast`. Reading the Ethereum address works fine using `cast`, using the same above command, once I launch the Ethereum app. However, `eip712sign` fails with the same above error.  I recently updated my Nano S Plus firmware to 1.3.0 in an attempt to fix the issue, but I'm unsure if this requirement is due to the firmware update.

**Environment Information:**

- Operating System: MacOS Sonoma 14.7.1 - Apple Silicon
- Foundry Versions Used: `143abd6a`, `e5dbb7a3`, and `626221f5`.
- Device: Ledger Nano S Plus

**Comment by geoknee** [comment]
Could this be related to [ops fail on ARM machines due to eip712signer incompatibility](https://github.com/ethereum-optimism/superchain-ops/issues/375) ?

**Comment by Inphi** [comment]
> Could this be related to [ops fail on ARM machines due to eip712signer incompatibility](https://github.com/ethereum-optimism/superchain-ops/issues/375) ?

I don't think so. The `eip712signer` works "fine" in the sense that it runs but is unable to recognize the ledger device. Whereas, I think the problem in that issue is that `eip712signer` would get killed by the OS because its binary target isn't expected.

**Comment by Inphi** [comment]
@mds1 It's a Ledger Nano S Plus.

**Comment by Ethnical** [comment]
Okay, I am currently with @Inphi and we tried multiples hardware things as restart, change with a low level terminal (terminal native of macos) also changing usb and usbc port.

I also add the same issue with @sebastianst this morning but I was thinking was related to only the **Flex**.
https://github.com/ethereum-optimism/superchain-ops/issues/395
still investigating

**Comment by mds1** [comment]
This is completed by https://github.com/ethereum-optimism/superchain-ops/pull/453
